File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int i = 0;
        int j = heights.length-1;
        int max = 0;
    	while(i < j) {

    		if(max < (j - i) * Math.min(heights[i],  heights[j])) {
    			max = Math.min(heights[i],  heights[j]) * (j-i);
//    			System.out.println("MAX: "+ max + "; i: " + i + "; j: " + j);
    		}
    		if(heights[i] > heights[j]) {
    			j--;
    		}else {
    			i++;
    		}
    	}
    	
    	return max;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int mx = 0;
        for(int i = 0; i < s.size(); i++){
            set<int> a;
            for(int j = i; j < s.size(); j++){
                if(a.contains(s[j])) break;
                a.insert(s[j]);
            }
            int k = a.size();
            mx = max(mx, k);
        }
        return mx;
    }
};
